app/vlinsert/journald/journald.go

```diff
@@ -273,30 +273,28 @@ func readJournaldLogEntry(streamName string, lr *insertutil.LineReader, lmp inse
 			fb.name = append(fb.name[:0], line...)
 			name = bytesutil.ToUnsafeString(fb.name)
 
-			fb.value = fb.value[:0]
-			for len(fb.value) < 8 {
-				if !lr.NextLine() {
-					if err := lr.Err(); err != nil {
-						return fmt.Errorf("cannot read value size: %w", err)
-					}
-					return fmt.Errorf("unexpected end of stream while reading value size")
+			if !lr.NextLine() {
+				if err := lr.Err(); err != nil {
+					return fmt.Errorf("cannot read value size: %w", err)
 				}
-				fb.value = append(fb.value, lr.Line...)
-				fb.value = append(fb.value, '\n')
+				return fmt.Errorf("unexpected end of stream while reading value size")
 			}
+			fb.value = append(fb.value[:0], lr.Line...)
+			fb.value = append(fb.value, '\n')
 			size := binary.LittleEndian.Uint64(fb.value[:8])
+			fb.value = fb.value[8:]
 
-			for size > uint64(len(fb.value[8:])) {
+			for size >= uint64(len(fb.value)) {
```

Comment:
If the `size` equals the length of the `fb.value`, then there is no need in reading the next line - we already finished reading the full binary-encoded value of the given size, which ends with a newline.

Reply:
left only this change and added tests for this case